src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1900,7 +1910,20 @@ private static Binder GetLambdaEnclosingBinder(int position, CSharpSyntaxNode st
         /// </summary>
         protected void EnsureNullabilityAnalysisPerformedIfNecessary()
         {
+#if !DEBUG
+            // In release mode, when the semantic model options include DisableNullableAnalysis,
+            // we want to completely avoid doing any work for nullable analysis.
+#pragma warning disable RSEXPERIMENTAL001 // Internal usage of experimental API
+            if (DisableNullableAnalysis)
```

Comment:
This appears to be checked by `IsNullableAnalysisEnabled`, why the separate check here?

Reply:
There is an existing behavior in release mode where, if `IsNullableAnalysisEnabled()` is false and `Compilation.IsNullableAnalysisEnabledAlways` is true, we performed full binding and nullable analysis, and used the results in the semantic model.

We are doing this check to establish the behavior for the new option, to not do any of this work and not use any analysis results, regardless of any other pre-existing nullable enabling/disabling options, without introducing a behavior change for existing scenarios.